--- a/include/slam_toolbox/get_pose_helper.hpp
+++ b/include/slam_toolbox/get_pose_helper.hpp
@@ -41,22 +41,16 @@ public:

   bool getOdomPose(karto::Pose2 & karto_pose, const rclcpp::Time & t)
   {
-    geometry_msgs::msg::TransformStamped base_ident, odom_pose;
-    base_ident.header.stamp = t;
-    base_ident.header.frame_id = base_frame_;
-    base_ident.transform.rotation.w = 1.0;
-
     try {
-      odom_pose = tf_->transform(base_ident, odom_frame_);
+      auto odom_pose = tf_->lookupTransform(
+        odom_frame_, base_frame_, tf2::TimePointZero);
+      const double yaw = tf2::getYaw(odom_pose.transform.rotation);
+      karto_pose = karto::Pose2(odom_pose.transform.translation.x,
+          odom_pose.transform.translation.y, yaw);
+      return true;
     } catch (tf2::TransformException & e) {
       return false;
     }
-
-    const double yaw = tf2::getYaw(odom_pose.transform.rotation);
-    karto_pose = karto::Pose2(odom_pose.transform.translation.x,
-        odom_pose.transform.translation.y, yaw);
-
-    return true;
   }

 private:
